backends/dpdk/dpdkArch.cpp

```diff
@@ -239,6 +253,166 @@ const IR::Node *ConvertToDpdkArch::preorder(IR::Member *m) {
 
 
 
+const IR::Node *ConvertLookahead::preorder(IR::AssignmentStatement *statement) {
+    if (!statement->right->is<IR::MethodCallExpression>())
+        return statement;
+    auto mce = statement->right->to<IR::MethodCallExpression>();
+
+    if (mce->type->is<IR::Type_Header>())
+        return statement;
+
+    auto mi = P4::MethodInstance::resolve(mce, refMap, typeMap);
+    if (!mi->is<P4::ExternMethod>())
+        return statement;
+    auto em = mi->to<P4::ExternMethod>();
+    if (em->originalExternType->name != P4::P4CoreLibrary::instance.packetIn.name ||
+            em->method->name != P4::P4CoreLibrary::instance.packetIn.lookahead.name)
+        return statement;
+
+    IR::ID newHeaderFieldName;
+    const IR::Type *newHeaderFieldType = nullptr;
+
+    /**
+     * Thanks to the fact that lookaheads are already expanded by P4::ExpandLookahead pass
+     * and structures are already flattened, following cases should cover all possibilities.
+     */
+    if (auto left = statement->left->to<IR::PathExpression>()) {
+        /**
```

Comment:
The actual expression on the LHS does not really matter. Since you know that the header you synthesize has a single field you can always use the same field name. So this code can be simplified. You do need fresh names for the type and the variable.

Reply:
done in 91c72491f4eb19f08fe83e1c4140e03aee2cfc1f